packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Timer } from "@fluidframework/common-utils";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ISummarizeHeuristicData, ISummarizeHeuristicRunner, ISummaryAttempt } from "./summarizerTypes";
+import { SummarizeReason } from "./summaryGenerator";
+
+/** Simple implementation of class for tracking summarize heuristic data. */
+export class SummarizeHeuristicData implements ISummarizeHeuristicData {
+    protected _lastAttempt: ISummaryAttempt;
+    public get lastAttempt(): ISummaryAttempt {
+        return this._lastAttempt;
+    }
+
+    protected _lastSuccessfulSummary: Readonly<ISummaryAttempt>;
+    public get lastSuccessfulSummary(): Readonly<ISummaryAttempt> {
+        return this._lastSuccessfulSummary;
+    }
+
+    constructor(
+        public lastOpSequenceNumber: number,
+        attemptBaseline: ISummaryAttempt,
+    ) {
+        this._lastAttempt = attemptBaseline;
+        this._lastSuccessfulSummary = { ...attemptBaseline };
+    }
+
+    public initialize(lastSummary: Readonly<ISummaryAttempt>) {
```

Comment:
It's a little weird that `_lastAttempt` and `_lastSuccessfulSummary` are assigned in the constructor as well as here. This seems more like an `update` than intialize.

Reply:
True, I thought about that as well. The intent is that it's only called once to initialize the data, then subsequent attempts would be done via `recordAttempt` and acks by `markLastAttemptAsSuccessful`.

I kind of agree that it could be better / more general, but I figure this most closely (and simply) matches the current use, and since it's not yet exposed externally, it would be easy to generalize when we decide to do so.